File: drills/ts/solutions/nc34_median_two_sorted__concat_sort.ts
export function findMedianSortedArrays(nums1: number[], nums2: number[]): number {
  const merged = [...nums1, ...nums2].sort((a, b) => a - b);
  if (merged.length === 0) return 0;
  // One expression for both parities: for an odd length the two indices are the
  // same element, so the average of it with itself is itself.
  return (merged[Math.floor(merged.length / 2)] + merged[Math.floor((merged.length - 1) / 2)]) / 2;
}

File: drills/go/solutions/nc34_median_two_sorted__concat_sort.go
package main

import "sort"

func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	all := append(append([]int{}, nums1...), nums2...)
	sort.Ints(all)
	n := len(all)
	if n%2 == 1 {
		return float64(all[n/2])
	}
	return float64(all[n/2-1]+all[n/2]) / 2
}
